### PR TITLE
[in_app_purchase] Add documentation for price change confirmations

### PR DESCRIPTION
**Pull request by renefloor**
The Android and iOS plugins for in_app_purchase added features to handle price change confirmations. This PR adds the explanation to the readme of the generic package.

This PR is a WIP/Draft as the iOS feature is not completely finished yet.

Android: PR #4077 and issue flutter/flutter#85038
iOS PR #4085 and issue flutter/flutter#85129

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

File: packages/in_app_purchase/in_app_purchase/README.md [hunk 1]
```diff
@@ -247,6 +247,55 @@ InAppPurchase.instance
     .buyNonConsumable(purchaseParam: purchaseParam);
 ```
 
+### Confirming subscription price changes
+
+When you change the price of a subscription the user will need to confirm the price change. If the user does not 
+confirm the price change the subscription will not be auto-renewed. On both iOS and Android the user will 
+automatically get a popup to confirm the price change, but you as a developer can show the popup on a moment that 
+suits you better. This works different on the Apple App Store and on the Google Play Store.
```
Comment (mvanbeusekom) [inline packages/in_app_purchase/in_app_purchase/README.md:255]:
```suggestion
When the price of a subscription is changed the consumer will need to confirm that price change. If the consumer does not 
confirm the price change the subscription will not be auto-renewed. By default on both iOS and Android the consumer will 
automatically get a popup to confirm the price change, but App developers can override this mechanism and show the popup on a later moment so it doesn't interrupt the critical flow of the App. This works different on the Apple App Store and on the Google Play Store.
```

File: packages/in_app_purchase/in_app_purchase/README.md [hunk 2]
```diff
@@ -247,6 +247,55 @@ InAppPurchase.instance
...
+
+After 7 days the user will be notified through email and notifications on Google Play to agree with the new price, 
+so you have 7 days to explain the user in your app that the price is going to change and ask them to accept the 
+change. You will have to keep track of whether or not the price change is already accepted in your app or in the 
+backend. The [Google Play API](https://developers.google.com/android-publisher/api-ref/rest/v3/purchases.subscriptions) 
+can be used to check whether or not the price change is accepted by the user by reading the priceChange property on 
+a subscription object.
+
+The InAppPurchaseAndroidPlatformAddition can be used to show the price change confirmation flow. The additions 
+contain the function `launchPriceChangeConfirmationFlow` which needs the sku code of the subscription.
```
Comment (mvanbeusekom) [inline packages/in_app_purchase/in_app_purchase/README.md:271]:
```suggestion
When the subscription price is raised, the consumer should approve the price change within 7 days. The official 
documentation can be found [here](https://support.google.com/googleplay/android-developer/answer/140504?hl=en#zippy=%2Cprice-changes).
When the price is lowered the consumer will automatically receive the lower price and does not have to approve the price change.

After 7 days the consumer will be notified through email and notifications on Google Play to agree with the new price. App developers have 7 days to explain the consumer that the price is going to change and ask them to accept this change. App developers have to keep track of whether or not the price change is already accepted within the app or in the backend. The [Google Play API](https://developers.google.com/android-publisher/api-ref/rest/v3/purchases.subscriptions) can be used to check whether or not the price change is accepted by the consumer by reading the `priceChange` property on a subscription object.

The `InAppPurchaseAndroidPlatformAddition` can be used to show the price change confirmation flow. The additions contain the function `launchPriceChangeConfirmationFlow` which needs the SKU code of the subscription. 
```

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Rewrote the description to remove the "you" to make it more clear who is responsible for which actions.

File: packages/in_app_purchase/in_app_purchase/example/lib/main.dart [hunk 3]
```diff
@@ -438,6 +451,28 @@ class _MyAppState extends State<_MyApp> {
...
+              .getPlatformAddition<InAppPurchaseAndroidPlatformAddition>();
+      var priceChangeConfirmationResult =
+          await androidAddition.launchPriceChangeConfirmationFlow(
+        sku: 'purchaseId',
+      );
+      if (priceChangeConfirmationResult.responseCode == BillingResponse.ok) {
+        // TODO acknowledge price change
+      } else {
+        // TODO show error
+      }
```
Comment (renefloor) [inline packages/in_app_purchase/in_app_purchase/example/lib/main.dart:467]: @mvanbeusekom I'm not sure what to do here. Should I just add a print statement, leave these todo's or do you have an idea?

Comment (mvanbeusekom) [reply packages/in_app_purchase/in_app_purchase/example/lib/main.dart:467]: Maybe we can simply show a snackbar message confirming the price change or showing the error?

**Review by mvanbeusekom — COMMENTED** [review]
Look good % the `// TODO: ...` comments in the Android specific code (see also my reply to your earlier comment).

**Comment by renefloor** [comment]
@mvanbeusekom I also added the code that actually calls the price change function: https://github.com/flutter/plugins/pull/4092/commits/4b62107df09d8ed42b7676de791ada30640c48de#diff-4a28c57f2e830b968dfb575e9152d91d12d3d276641696552e79df589daa1291L261-R263

**Review by mvanbeusekom — APPROVED** [review]
LGTM

**Pull request merged**